File: SWEA/D3/5215. 햄버거 다이어트/햄버거 다이어트.py
T = int(input())
for tc in range(1, T+1):
	n, k = map(int, input().split())
	dp = [[0 for _ in range(k+1)] for _ in range(n+1)]
	for i in range(1, n+1):
		v, w = map(int, input().split())
		for j in range(k):
			if j-w >= 0:
				dp[i][j] = dp[i-1][j-w] + v
			dp[i][j] = max(dp[i-1][j], dp[i][j])
	
	print(f'#{tc} {max(dp[-1])}')

File: SWEA/D3/5215. 햄버거 다이어트/햄버거 다이어트.java
import java.io.*;
import java.util.*;

public class Solution {
    static BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
    static StringBuilder sb = new StringBuilder();
    static StringTokenizer st;

    public static void main(String[] args) throws Exception {
    	int T = Integer.parseInt(input.readLine());
    	for (int t = 1; t<=T; t++) {
    		st = new StringTokenizer(input.readLine());
    		int n = Integer.parseInt(st.nextToken());
    		int k = Integer.parseInt(st.nextToken());
    		
    		int[][] dp = new int[n+1][k+5];
    		int ans = 0;
    		
    		for (int i=1; i<=n; i++) {
    			st = new StringTokenizer(input.readLine());
    			int v = Integer.parseInt(st.nextToken());
    			int w = Integer.parseInt(st.nextToken());
    			
    			for (int j=0; j<k; j++) {
    				if (j-w >= 0)
    					dp[i][j] = dp[i-1][j-w] + v;
    				dp[i][j] = Math.max(dp[i-1][j], dp[i][j]);
    				ans = Math.max(dp[i][j], ans);
    			}
    		}
    		
    		sb.append("#").append(t).append(" ").append(ans).append("\n");
    	}
    	System.out.println(sb.toString());
    }
}
